coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,97 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). The fileproxy object also protects your file from being
```

Comment:
Yes it can still edit the file using `open()`. What I wanted to say was it protects the file contents loaded into the fileproxy by having immutable objects and private class members so that other bears get the same content.